core/src/main/java/io/kestra/core/runners/RunContext.java

```diff
@@ -434,12 +434,22 @@ public RunContext forWorkerDirectory(ApplicationContext applicationContext, Work
         return this;
     }
 
+    // Default behavior
     public String render(String inline) throws IllegalVariableEvaluationException {
-        return variableRenderer.render(inline, this.variables);
+        return variableRenderer.render(inline, this.variables, true);
+    }
+
+    public String render(String inline, boolean renderVariables) throws IllegalVariableEvaluationException {
```

Comment:
```suggestion
    public String render(String inline, boolean preprocessVariables) throws IllegalVariableEvaluationException {
```

I think it's better to call it preprocessed as it's what you do and variable are rendered anyway as we are recursive.